src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,302 @@
+  static async run(flattenMap: any) {
+    /**
+     * 1. Create metadata if pathToHash exists but does not have actual metadata files,
+     *      If pathToHash exists and there is a actual metadata file, but there is no path inside,
+     * create a path and data insde
+     */
+    await this.createMetadata(flattenMap);
+
+    // 2. Replace is_deleted with true for all metadata not in pathToHash
+    await this.deleteMetadata(flattenMap);
+  }
+
+  static async createMetadata(flattenMap: any) {
+    for (let path in flattenMap) {
+      const hash = flattenMap[path];
+      if (vscode.workspace.workspaceFolders) {
+        await Metadata.createDefault(
+            vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path), hash);
+      }
+    }
+  }
+
+  static async deleteMetadata(flattenMap: any) {
+    if (vscode.workspace.workspaceFolders === undefined ||
+        !fs.existsSync(
+            vscode.Uri
+                .joinPath(vscode.workspace.workspaceFolders[0].uri, '.one-vscode/info/hash_objects')
+                .fsPath)) {
+      return;
+    }
+    const baseUri = vscode.Uri.joinPath(
+        vscode.workspace.workspaceFolders[0].uri, '.one-vscode/info/hash_objects');
+    const files = await vscode.workspace.fs.readDirectory(baseUri);
+    for (const file of files) {
+      const hashFolderUri = vscode.Uri.joinPath(baseUri, file[0]);
+      const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+      for (const hashFile of hashList) {
+        const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+        let metaObj =
+            JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+        const hash = file[0] + hashFile[0].split('.')[0];
+        for (const path in metaObj) {
+          if (flattenMap[path] === undefined) {
+            const uri = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path);
+            await Metadata.disable(uri, hash);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * A Singleton Object
+ *
+ * PathToHash is a data structure that stores the path of the actual data
+ * and the hash value generated by the contents of the actual data
+ * to synchronize the actual data and metadata
+ */
+
+export class PathToHash {
+  private static _instance: PathToHash;
+  private _map: any;
+  public static startFlag: boolean = false;
+
+  private constructor() {}
+
+  public static async getInstance() {
+    if (!this._instance) {
+      this._instance = new PathToHash();
+      this._instance._map = await this._instance.init();
+      PathToHash.startFlag = true;
+    }
+    return this._instance;
+  }
+
+  private async init() {
+    if (vscode.workspace.workspaceFolders === undefined) {
+      return;
+    }
+    const uri = vscode.workspace.workspaceFolders[0].uri;
+    const map = await this.scanRecursively(uri);
+    await MetadataSynchronizer.run(await this.getFlatMap(map));
+    return map;
+  }
+
```

Comment:
```suggestion

  /**
   * Scan the file system and generates hash of all the files.
   * The map is structured as a tree of directories and files.
   * 
   * @example 
   *  uri: 'A/B/C.txt'
   *  return: {'A': {'B': {'C.txt': <C.txt's hash value> } } } }
   */
```

How about adding the comment to you explain the function?